src/config.c

```diff
@@ -173,54 +163,151 @@ config_read_cmd(const char *ipath, const char *set_command_format)
     if (path == NULL) return NULL;
    }
 
-  set_command_size = snprintf(NULL,0,set_command_format,path)+1;
-  set_command = (char *)malloc(set_command_size);
-  if (set_command == NULL)
+  rha = config_new(path);
+  free(path);
+
+  if ( ! config_read_cmd(rha->config_path, set_command_format, rha) )
    {
-    free(path);
+    config_free(rha);
     return NULL;
    }
-  snprintf(set_command, set_command_size, set_command_format, path);
 
-  cf = popen(set_command, "r");
-  if (cf == NULL) 
+  config_dir = (char *)malloc(strlen(rha->config_path)+3);
+  sprintf(config_dir, "%s.d", rha->config_path);
+  file_cnt = scandir(config_dir, &file_list, NULL, alphasort);
+  for (i = 0; i < file_cnt; i++)
    {
-    free(set_command);
-    free(path);
-    return NULL;
+    if (file_list[i]->d_name[0] != '.')
+	 {
+      snprintf(full_file, PATH_MAX, "%s/%s", config_dir, file_list[i]->d_name);
+      config_read_cmd(full_file, set_command_format, rha);
+     }
+    free(file_list[i]);
    }
+  free(file_list);
 
-  free(set_command);
+  pwd = getpwuid(geteuid());
+  if (pwd != NULL && pwd->pw_dir != NULL)
+   {
+    snprintf(full_file, PATH_MAX, "%s/.blah/user.config", pwd->pw_dir);
+    test = fopen(full_file, "r");
+	if (test != NULL)
+     {
+      fclose(test);
```

Comment:
There's a stray tab here...

```suggestion
    test = fopen(full_file, "r");
    if (test != NULL)
     {
      fclose(test);
```

Alternatively you could just use access(2) instead of fopen/fclose:

```suggestion
    if (access(full_file, R_OK) == 0)
     {
```

Reply:
The fopen()/fclose() is original blahp code that was moved. access() is definitely better, so I'll add that change.